stock_location_orderpoint/models/stock_location_orderpoint.py

```diff
@@ -0,0 +1,429 @@
+        )
+        return {
+            self.env["stock.location"]
+            .browse(res["location_id"][0]): self.env["stock.move"]
+            .browse(res["ids"])
+            for res in moves_grouped
+        }
+
+    def _sort_orderpoints(self):
+        return self.sorted()
+
+    @api.model
+    def _compute_quantities_dict(self, locations, products):
+        qties = {}
+        for location in locations:
+            qties_on_location = qties.setdefault(location, {})
+            products = products.with_context(location=location.id)
+            for product_id, qties_dict in products._product_available().items():
+                product = products.browse(product_id)
+                qties_on_location[product] = qties_dict
+        return qties
+
+    def _get_qty_to_replenish(
+        self, product, qties_on_locations, qty_already_replenished=0
+    ):
+        """
+        Returns a qty to replenish for a given orderpoint and product
+        Calls the _get_qty_to_replenish_{self.replenish_method}
+        To add a new rule how the qty_to_replenish is computed
+        simple add a new
+         - value in replenish_method selection
+         - method called _get_qty_to_replenish_<your_new_replenish_method>
+        """
+        self.ensure_one()
+        product.ensure_one()
+
+        qty_to_replinsh_func_name = f"_get_qty_to_replenish_{self.replenish_method}"
+        if not hasattr(self, qty_to_replinsh_func_name):
+            return 0
+
+        return getattr(self, qty_to_replinsh_func_name)(
+            product, qties_on_locations, qty_already_replenished
+        )
+
+    def _get_qty_to_replenish_fill_up(
+        self, product, qties_on_locations, qty_already_replenished=0
+    ):
+        if not self.location_src_id:
+            return 0
+
+        qties_on_dest = qties_on_locations[self.location_id][product]
+        virtual_available_on_dest = qties_on_dest["virtual_available"]
+        if (
+            float_compare(
+                virtual_available_on_dest, 0, precision_rounding=product.uom_id.rounding
+            )
+            >= 0
+        ):
+            return 0
+
+        virtual_available_on_dest = abs(virtual_available_on_dest)
+        qties_on_src = qties_on_locations[self.location_src_id][product]
+        virtual_available_on_src = (
+            qties_on_src["virtual_available"] - qties_on_src["incoming_qty"]
+        )
+        if (
+            float_compare(
+                virtual_available_on_src,
+                0,
+                precision_rounding=product.uom_id.rounding,
+            )
+            <= 0
+        ):
+            return 0
+
+        qty_to_replenish = virtual_available_on_dest - qty_already_replenished
+        return min(qty_to_replenish, virtual_available_on_src)
+
+    def _get_qties_to_replenish(self, moves_by_location):
+        products = self.env["product.product"]
+        for moves in moves_by_location.values():
+            products |= moves.product_id
```

Comment:
As `|` is underperformant in ORM:

```suggestion
            product_ids.append(moves.product_id.id)
            qties_on_locations = self._compute_quantities_dict(
            (self.location_id | self.location_src_id), self.env["product.product"].browse(product_ids)
        )
```